app/controllers/shared/api.php

```diff
@@ -19,234 +19,267 @@
 use Utopia\Database\Validator\Authorization;
 use Utopia\Registry\Registry;
 
-App::init(function (App $utopia, Request $request, Response $response, Document $project, Document $user, Event $events, Audit $audits, Mail $mails, Stats $usage, Delete $deletes, EventDatabase $database, Database $dbForProject, string $mode) {
-
-    $route = $utopia->match($request);
-
-    if ($project->isEmpty() && $route->getLabel('abuse-limit', 0) > 0) { // Abuse limit requires an active project scope
-        throw new Exception('Missing or unknown project ID', 400, Exception::PROJECT_UNKNOWN);
-    }
-
-    /*
-     * Abuse Check
-     */
-    $abuseKeyLabel = $route->getLabel('abuse-key', 'url:{url},ip:{ip}');
-    $timeLimitArray = [];
-
-    $abuseKeyLabel = (!is_array($abuseKeyLabel)) ? [$abuseKeyLabel] : $abuseKeyLabel;
-
-    foreach ($abuseKeyLabel as $abuseKey) {
-        $timeLimit = new TimeLimit($abuseKey, $route->getLabel('abuse-limit', 0), $route->getLabel('abuse-time', 3600), $dbForProject);
-        $timeLimit
-            ->setParam('{userId}', $user->getId())
-            ->setParam('{userAgent}', $request->getUserAgent(''))
-            ->setParam('{ip}', $request->getIP())
-            ->setParam('{url}', $request->getHostname() . $route->getPath());
-        $timeLimitArray[] = $timeLimit;
-    }
-
-    $closestLimit = null;
-
-    $roles = Authorization::getRoles();
-    $isPrivilegedUser = Auth::isPrivilegedUser($roles);
-    $isAppUser = Auth::isAppUser($roles);
-
-    foreach ($timeLimitArray as $timeLimit) {
-        foreach ($request->getParams() as $key => $value) { // Set request params as potential abuse keys
-            if (!empty($value)) {
-                $timeLimit->setParam('{param-' . $key . '}', (\is_array($value)) ? \json_encode($value) : $value);
-            }
+App::init()
+    ->groups(['api'])
```

Comment:
The rest of PR puts groups under injection. Let's do it here too for consistency.

Reply:
I think groups should be at the top.